libs/k8s-client/tests/ApiClient/EventsApiClientFunctionalTest.php

```diff
@@ -5,96 +5,60 @@
-use Keboola\K8sClient\RetryProxyFactory;
+use Kubernetes\API\Event as EventsApi;
 use Kubernetes\Model\Io\K8s\Api\Core\V1\Event;
-use Kubernetes\Model\Io\K8s\Api\Core\V1\Pod;
-use Kubernetes\Model\Io\K8s\Apimachinery\Pkg\Apis\Meta\V1\DeleteOptions;
-use Monolog\Logger;
 use PHPUnit\Framework\TestCase;
 
 class EventsApiClientFunctionalTest extends TestCase
 {
-    private static KubernetesApiClientFacade $apiClient;
-    private static EventsApiClient $eventsApiClient;
+    /**
+     * @template-use BaseNamespaceApiClientTestCase<EventsApi, EventsApiClient>
+     */
+    use BaseNamespaceApiClientTestCase;
 
-    public static function setUpBeforeClass(): void
+    public function setUp(): void
     {
-        parent::setUpBeforeClass();
-
-        $logger = new Logger('test');
-
-        $clientFacade = (new GenericClientFacadeFactory(
-            (new RetryProxyFactory($logger))->createRetryProxy(),
-            $logger
-        ))->createClusterClient(
-            (string) getenv('K8S_HOST'),
-            (string) getenv('K8S_TOKEN'),
-            (string) getenv('K8S_CA_CERT_PATH'),
-            (string) getenv('K8S_NAMESPACE'),
+        parent::setUp();
+        $this->setUpBaseNamespaceApiClientTest(
+            EventsApi::class,
+            EventsApiClient::class,
         );
-
-        self::$apiClient = $clientFacade;
-        self::$eventsApiClient = $clientFacade->events();
-
-        // generate a few events to test below
-        self::cleanupCluster();
-
-        $clientFacade->pods()->create(new Pod([
-            'metadata' => [
-                'name' => 'test-pod',
-                'labels' => [
-                    'app' => 'EventsApiClientFunctionalTest',
-                ],
-            ],
-            'spec' => [
-                'containers' => [
-                    [
-                        'name' => 'nginx',
-                        'image' => 'nginx',
-                    ],
-                ],
-            ],
-        ]));
-    }
-
-    public static function tearDownAfterClass(): void
-    {
-        parent::tearDownAfterClass();
-        self::cleanupCluster();
     }
 
-    private static function cleanupCluster(): void
+    protected function createResource(array $metadata): Event
     {
-        self::$apiClient->deleteAllMatching(
-            new DeleteOptions(['gracePeriodSeconds' => 0]),
-            ['labelSelector' => 'app=EventsApiClientFunctionalTest']
-        );
+        return new Event([
+            'metadata' => $metadata,
+            'type' => 'Normal',
+            'reason' => 'Testing',
+            'message' => 'EventsApiClientFunctionalTest event',
+            'involvedObject' => [
+                'kind' => 'ConfigMap',
+                'name' => 'kube-root-ca.crt',
+                'namespace' => (string) getenv('K8S_NAMESPACE'),
+            ],
+        ]);
     }
 
-    public function testList(): void
+    public function testListForAllNamespaces(): void
```

Comment:
Tady bych potřeboval další namespace a nagrantovat si pro něj oprávnění. To mi příjde jako overkill. Udělal jsem alespoň unit test 🤷 

Původně jsem měl test udělanej tak, že jsem spoléhal na to že už tam nějaké eventy v jiných namespacech jsou. Ale to fungovalo jen u AKS. V EKS žádný nebyly :)

Reply:
Jj.. ono ty eventy maji defaultne retenci tusim 1 hodinu, takze pokud se na clusteteru chvili nic nedeje (coz na dev/ci muze byt), muze to byt celkem prazdny.

Souhlas, ze v tomhle pripade unit test je enough. Pokud by ti to nakonec hodne nedalo spat, muzeme tu metodu nejspis i odstranit uplne, protoze nakonec se nepouzila. A cross-namespace testy doresit az jindy, pokud by to nekdy v budoucnu bylo potreba resit.